### PR TITLE
Fixes #1189, Braille Spaces & Empty Paragraphs removal

### LINKED ISSUE #1189: Remove empty whitespace-only blocks from posts
I feel like @agarcia-caicedo mentioned this in another issue, but I can't find it now.  A number of blog platforms add extra whitespace blocks to posts, for example:

Screen Shot 2020-10-12 at 4 56 27 PM

We could try doing some work in our HTML post-processing step to remove things like empty `<p>`, multiple `<br>`, `empty `<div></div>` or `<div><p></p></div>`

### PR DESCRIPTION
**Pull request by Metropass**
## Issue This PR Addresses
Fixes #1189, and also added a check for the 0-dot braille character (Unicode Hex: 2800).
For the `<br>` removal, I had to split the work in half, as there's an issue with DOM not properly cutting the extra `/n` values. 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,6 +75,8 @@
   "dependencies": {
     "@elastic/elasticsearch": "7.9.0",
     "@elastic/elasticsearch-mock": "0.3.0",
+    "@next/mdx": "10.0.6",
+    "@octokit/webhooks": "8.0.2",
```
Comment (manekenpix) [inline package.json:79]: I don't think these are used in this PR.

**Review by manekenpix — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Metropass) [reply package.json:79]: It's not, but I had to include them, otherwise it'll fail the tests for missing packages 🤷

Comment (manekenpix) [reply package.json:79]: how can you fail tests if those packages are not used in master?

Comment (manekenpix) [reply package.json:79]: oh, I see that `@next/mdx` is already included in master, and here it seems you're updating it to version `10.0.6`.

Comment (manekenpix) [reply package.json:79]: Also, I ran our tests without `@octokit/webhooks` and it didn't fail.
Should we update `@next/mdx` in  a different PR maybe?

cc: @humphd

Comment (humphd) [reply package.json:79]: Yes, cc @chrispinkney, who is managing this.  Let's not do it here, please revert this and we'll do it in a new PR.

File: src/backend/utils/html/index.js [hunk 2]
```diff
@@ -42,7 +44,10 @@ module.exports = function process(html) {
   lazyLoad(dom);
   // Replace <code> elements with encoded entities to use characters
   replaceCodeEntities(dom);
-
+  // Remove empty <p> elements
+  removeEmptyParagraphs(dom);
   // Return the resulting HTML
```
Comment (humphd) [inline src/backend/utils/html/index.js:49]: This comment doesn't match what you're doing on the next line.  Move this comment to 51 above the final return statement, and properly document what you're doing with this code.

File: src/backend/utils/html/remove-zero-dot-braille.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = function (dom) {
+  const regexValue = new RegExp('u{2800}');
```
Comment (humphd) [inline src/backend/utils/html/remove-zero-dot-braille.js:2]: To include unicode in JS Regex, you need to use `\u2800`.  Also, let's do it inline below, and use a global replace to get all instances:

```js
    if (!paragraphInnerHTML.replace(/\u2800/g, '').trim()) {
```

File: test/remove-zero-dot-braille.test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const toDom = require('../src/backend/utils/html/dom');
+const removeBraille = require('../src/backend/utils/html/remove-zero-dot-braille');
+
+test('should remove <p>⠀⠀⠀⠀</p> (braille)', () => {
+  const htmlData = toDom('<div><p></p></div>');
```
Comment (humphd) [inline test/remove-zero-dot-braille.test.js:5]: Be more explicit, use Unicode hex entity: `&#x2800;`

File: test/remove-empty-paragraphs.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  test('should remove <p>   </p> (spaces)', () => {
+    const htmlData = toDom('<div><p>   </p></div>');
+    removeEmptyParagraphs(htmlData);
+
+    const expectedHtml = '<div></div>';
+    expect(htmlData.window.document.body.innerHTML).toEqual(expectedHtml);
+  });
+
+  test('should remove <p>     </p> (tabs)', () => {
```
Comment (humphd) [inline test/remove-empty-paragraphs.test.js:21]: Can you be more explicit with the inclusion of the tabs here? `&#9;` will do it.

**Review by humphd — CHANGES_REQUESTED** [review]
Left some feedback.

Also, a question: why are you focused on braille-dots specifically?  There are lots of invisible whitespace characters.  Why not get rid of them all with something like https://github.com/junosuarez/clean-whitespace instead?

**Comment by Metropass** [comment]
Closed. Added a better functioning PR at #1787

**Pull request closed without merging**